### PR TITLE
Add getters for client hello raw data

### PR DESCRIPTION
**Pull request by jrouviere**

### CODE REVIEW

**Review by rushilmehra — APPROVED** [review]

**Comment by rushilmehra** [comment]
lgtm, @nox want to take a quick look?

File: boring/src/ssl/mod.rs [hunk 1]
```diff
@@ -2240,6 +2240,16 @@ impl ClientHello<'_> {
     pub fn version_str(&self) -> &'static str {
         self.ssl().version_str()
     }
+
+    /// Returns the client hello raw data
+    pub fn raw_client_hello(&self) -> &[u8] {
```
Comment (nox) [inline boring/src/ssl/mod.rs:2245]: This is the entire client hello record, right? Then I would just name that as `as_bytes`.

File: boring/src/ssl/mod.rs [hunk 2]
```diff
@@ -2240,6 +2240,16 @@ impl ClientHello<'_> {
...
+        unsafe { slice::from_raw_parts(self.0.client_hello, self.0.client_hello_len) }
+    }
+
+    /// Returns the client random raw data
+    pub fn raw_random(&self) -> &[u8] {
```
Comment (nox) [inline boring/src/ssl/mod.rs:2250]: I think `random` is a better name, "raw" feels weird to me as there could not be a non-raw version of this, right?

File: boring/src/ssl/mod.rs (hunk 1 above)
Comment (jrouviere) [reply boring/src/ssl/mod.rs:2245]: Yes that's the whole message sent by the client. Agreed that's a better name, will change.

File: boring/src/ssl/mod.rs (hunk 2 above)
Comment (jrouviere) [reply boring/src/ssl/mod.rs:2250]: Good point, not much you can parse, I'll rename it.

**Review by nox — APPROVED** [review]

**Pull request merged**